kube_obj_parser.py

```diff
@@ -404,22 +407,27 @@ def parse(self, url, endpoint):
                         continue
 
     def _kube_get(self, url, endpoint):
+        headers = {}
+        k8s_cert_existed = False
+        if os.path.exists(K8S_BEARER_TOKEN_FILE_NAME) and os.stat(K8S_BEARER_TOKEN_FILE_NAME).st_size > 0:
+            with open(K8S_BEARER_TOKEN_FILE_NAME, 'r') as tokenfile:
+                headers = {'Authorization': 'Bearer ' + tokenfile.read() }
+        else:
+            Logger.log('Connect Kubernetes API server failed: Could not find bearer token at ' + K8S_BEARER_TOKEN_FILE_NAME + '. Exiting.')
+            sys.exit(1)
+        if os.path.exists(K8S_CA_CRT_FILE_NAME) and os.stat(K8S_CA_CRT_FILE_NAME).st_size > 0:
+            k8s_cert_existed = True
+
         if url:
-            return requests.get(url + endpoint)
+            if k8s_cert_existed:
+                return requests.get(url + endpoint, verify = K8S_CA_CRT_FILE_NAME, headers=headers)
```

Comment:
If I'm reading the code right, `verify` should be set to `True`, right?

Reply:
No, Here we check if the Kubernetes default cert `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`  exists, if it does exist, then we will set the `verify` to then path of the cert.

Reply:
Got it! I actually double-checked the specs and I thought `verify` had to be a boolean, which is not the case (https://requests.readthedocs.io/en/master/api/#requests.request). My bad.